### PR TITLE
🌱 Used path filters in GitHub actions to avoid throttling 

### LINKED ISSUE #2052: Use https://github.com/dorny/paths-filter for GH Actions
**Is your feature request related to a problem? Please describe.**
Right now our GH Actions run for every change and we are being throttled by GH API. We could use this https://github.com/dorny/paths-filter to filter out when each of these actions runs based on code changes.

### PR DESCRIPTION
**Pull request by Shofiya2003**
#### What kind of change does this PR introduce?
Implemented conditional run of jobs in GitHub Actions so that we are not throttled by GH API

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
Right now our GH Actions run for every change and we are being throttled by GH API

#### What is the new behavior (if this is a feature change)?**
Conditionally running GitHub actions according to the changes

- [ x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
Fixes #2052 

#### Special notes for your reviewer
Not included all the changes required, just implemented a single change for now.

#### Does this PR introduce a user-facing change?

```release-note
"NONE"
```

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -30,8 +30,24 @@ env:
...
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      runUnitTests: ${{ steps.filter.outputs.runUnitTests }} 
+    steps:
+    # For pull requests it's not necessary to checkout the code
+    - uses: dorny/paths-filter@v2
```
Comment (naveensrinivasan) [inline .github/workflows/main.yml:41]: Can you please pin it by SHA?

**Review by naveensrinivasan — COMMENTED** [review]
Thanks so much! Would you be able to give this a try in your fork and share the link here?

**Comment by Shofiya2003** [comment]
> Thanks so much! Would you be able to give this a try in your fork and share the link here?

How to do that could you please help me

File: .github/workflows/main.yml (hunk 1 above)
Comment (Shofiya2003) [reply .github/workflows/main.yml:41]: sure on it

**Comment by Shofiya2003** [comment]
> Thanks so much! Would you be able to give this a try in your fork and share the link here?

On it figured out how to do it

**Comment by Shofiya2003** [comment]
@naveensrinivasan I tried the changes in my fork by creating a PR, the job does not run when there are only md changes
Link to the PR 
https://github.com/Shofiya2003/scorecard/pull/1

File: .github/workflows/main.yml (hunk 1 above)
Comment (naveensrinivasan) [reply .github/workflows/main.yml:41]: Thanks!

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan I tried the changes in my fork by creating a PR, the job does not run when there are only md changes
> Link to the PR
> [Shofiya2003#1](https://github.com/Shofiya2003/scorecard/pull/1)

Apologize for the delay! Thank you!

File: .github/workflows/main.yml (hunk 1 above)
Comment (naveensrinivasan) [reply .github/workflows/main.yml:41]: A friendly reminder to pin by SHA

**Review by naveensrinivasan — COMMENTED** [review]
Thank you! LGTM. We would need the action to be pinned by SHA.

**Comment by naveensrinivasan** [comment]
@Shofiya2003 A friendly reminder.

**Comment by Shofiya2003** [comment]
@naveensrinivasan will surely do it today
Apologies for the delay

**Comment by Shofiya2003** [comment]
@naveensrinivasan I used https://github.com/mheap/pin-github-action has to be used to pin the action. Right?

**Comment by Shofiya2003** [comment]
@naveensrinivasan I tried to use rebase to DCO the commits. Is it alright? 
Should I create a new pull request?
Sorry, DCO and pinning the commits were new to me.

**Pull request closed without merging**